openapi.yaml

```diff
@@ -4,93 +4,285 @@ info:
       type: string
-    IncidentImpactType:
+      readOnly: true
+
+    IdList:
+      description: List of IDs for referencing other objects.
+      type: array
+      items:
+        $ref: '#/components/schemas/Id'
+
+    Incremental:
+      description: |
+        Positive and incrementing number for ordering and identfication of e.g. sub resources.
+      type: integer
+      readOnly: true
+
+    DisplayName:
+      description: Short and describing name.
       type: string
-    IncidentPhase:
+
+    Description:
+      description: A longer text for detailed informations.
       type: string
+
+    Date:
+      description: Point in time. Nullable for open ends.
+      type: string
+      format: date-time
+      nullable: true
+
+    # API objects
+    ImpactType:
+      type: object
+      properties:
+        id:
+          $ref: '#/components/schemas/Id'
+        displayName:
+          $ref: '#/components/schemas/DisplayName'
+        description:
+          $ref: '#/components/schemas/Description'
+      required:
+        - id
+
+    Impact:
+      description: An impact connects a component and an incident with an impact type.
+      type: object
+      properties:
+        type:
+          $ref: '#/components/schemas/ImpactType'
+        reference:
+          $ref: '#/components/schemas/Id'
+
+    ImpactList:
+      description: |
+        A list of impacts for a component or incident.
+        In case of an incident the IDs refer to a component and vice versa.
+      type: array
+      items:
+        $ref: '#/components/schemas/Impact'
+
+    Phase:
+      description: |
+        A single phase is just its name.
+        It can be referenced by its generation and order.
+        See: #/components/schemas/DisplayName
+      type: string
+
+    PhaseReference:
+      description: To reference a phase, its generation and order is needed.
+      type: object
+      properties:
+        displayName:
+          $ref: '#/components/schemas/DisplayName'
+        order:
+          $ref: '#/components/schemas/Incremental'
+        generation:
+          $ref: '#/components/schemas/Incremental'
+      required:
+        - order
+        - generation
+
+    PhaseList:
+      description: Phase resources are always handled as a list.
+      type: object
+      properties:
+        generation:
+          $ref: '#/components/schemas/Incremental'
+        phases:
+          type: array
+          items:
+            $ref: '#/components/schemas/Phase'
+      required:
+        - generation
+        - phases
+
     IncidentUpdate:
+      description: |
+        An update is a sub resource to an incident.
+        It's identified by the incident ID and its own order.
+        Updates happen in a given order.
       type: object
-      required: [text, createdAt]
       properties:
-        text:
-          type: string
+        order:
+          $ref: '#/components/schemas/Incremental'
+        displayName:
```

Comment:
An `IncidentUpdate` is having a `displayName`? Maybe rather `title` for `displayName` and `text` for `description`?

Reply:
I just wanted to keep the field naming analogous to the other objects. A display name is just a name to display. A title could be seen as a subtype of a display name.

Reply:
Keeping it as-is works for me :+1: